### PR TITLE
refactor ICache to chisel3.

### PR DESCRIPTION
**Pull request by SingularityKChen**
**Related issue**: 

**Type of change**: other enhancement

**Impact**: no functional change

**Development Phase**: implementation

**Release Notes**

Refactor ICache.scala to chisel3 style.

### CODE REVIEW

File: src/main/scala/rocket/ICache.scala [hunk 1]
```diff
@@ -574,21 +574,23 @@ class ICacheModule(outer: ICache) extends LazyModuleImp(outer)
...
+      data_array.write(mem_idx, VecInit(Seq.fill(nWays){dECC.encode(data)}), (0 until nWays).map(way === _.U))
     }
     // write access
     /** data read from [[data_array]]. */
     val dout = data_array.read(mem_idx, !wen && s0_ren)
     // Mux to select a way to [[s1_dout]]
     when (wordMatch(Mux(s1_slaveValid, s1s3_slaveAddr, io.s1_paddr))) {
       s1_dout := dout
+    }.otherwise {
+      s1_dout := VecInit(Seq.fill(nWays){0.U(dECC.width(wordBits).W)})
```
Comment (SingularityKChen) [inline src/main/scala/rocket/ICache.scala:586]: Not sure whether these two lines introduce the function bug.

File: src/main/scala/rocket/ICache.scala [hunk 2]
```diff
@@ -330,7 +330,7 @@ class ICacheModule(outer: ICache) extends LazyModuleImp(outer)
   val s0_vaddr = io.req.bits.addr
 
   /** valid signal for stage 1, drived by s0_valid.*/
-  val s1_valid = Reg(init=Bool(false))
+  val s1_valid = RegInit(init=false.B)
```
Comment (sequencer) [inline src/main/scala/rocket/ICache.scala:333]:
```suggestion
  val s1_valid = RegInit(false.B)
```

File: src/main/scala/rocket/ICache.scala [hunk 3]
```diff
@@ -437,18 +437,18 @@ class ICacheModule(outer: ICache) extends LazyModuleImp(outer)
     * indexed by (wayIndex ## setIndex)
     * after refill_done and not FENCE.I, (repl_way ## refill_idx) set to true.
     */
-  val vb_array = Reg(init=Bits(0, nSets*nWays))
+  val vb_array = RegInit(init=0.U(nSets*nWays))
```
Comment (sequencer) [inline src/main/scala/rocket/ICache.scala:440]:
```suggestion
  val vb_array = RegInit(init=0.U((nSets*nWays).W))
```

File: src/main/scala/rocket/ICache.scala [hunk 4]
```diff
@@ -437,18 +437,18 @@ class ICacheModule(outer: ICache) extends LazyModuleImp(outer)
...
   when (refill_one_beat) {
     accruedRefillError := refillError
     // clear bit when refill starts so hit-under-miss doesn't fetch bad data
     vb_array := vb_array.bitSet(Cat(repl_way, refill_idx), refill_done && !invalidated)
   }
 
   /** flush cache when invalidate is true. */
-  val invalidate = Wire(init = io.invalidate)
+  val invalidate = WireDefault(init = io.invalidate)
```
Comment (sequencer) [inline src/main/scala/rocket/ICache.scala:448]:
```suggestion
  val invalidate = WireDefault(io.invalidate)
```

File: src/main/scala/rocket/ICache.scala [hunk 5]
```diff
@@ -599,7 +600,7 @@ class ICacheModule(outer: ICache) extends LazyModuleImp(outer)
   /** ITIM index to access [[data_arrays]].
     * replace tag with way, word set to 0.
     */
-  val s2_scratchpad_word_addr = Cat(s2_hit_way, Mux(s2_slaveValid, s1s3_slaveAddr, io.s2_vaddr)(untagBits-1, log2Ceil(wordBits/8)), UInt(0, log2Ceil(wordBits/8)))
+  val s2_scratchpad_word_addr = Cat(s2_hit_way, Mux(s2_slaveValid, s1s3_slaveAddr, io.s2_vaddr)(untagBits-1, log2Ceil(wordBits/8)), 0.U(log2Ceil(wordBits/8)))
```
Comment (sequencer) [inline src/main/scala/rocket/ICache.scala:603]:
```suggestion
  val s2_scratchpad_word_addr = Cat(s2_hit_way, Mux(s2_slaveValid, s1s3_slaveAddr, io.s2_vaddr)(untagBits-1, log2Ceil(wordBits/8)), 0.U(log2Ceil(wordBits/8).W))
```

**Review by sequencer — CHANGES_REQUESTED** [review]

**Review by sequencer — APPROVED** [review]

**Pull request merged**